src/sif/autocost.cc

```diff
@@ -564,11 +593,23 @@ Cost AutoCost::TransitionCostReverse(const uint32_t idx,
     // Separate time and penalty when traffic is present. With traffic, edge speeds account for
     // much of the intersection transition time (TODO - evaluate different elapsed time settings).
     // Still want to add a penalty so routes avoid high cost intersections.
-    float seconds = turn_cost * edge->stopimpact(idx);
-    // Apply density factor penality if there isnt traffic on this edge or youre not using traffic
-    if (!edge->has_flow_speed() || flow_mask_ == 0)
-      seconds *= trans_density_factor_[node->density()];
+    float seconds = turn_cost;
+    bool is_turn = false;
+    if (edge->turntype(idx) == baldr::Turn::Type::kLeft ||
+        edge->turntype(idx) == baldr::Turn::Type::kSharpLeft ||
+        edge->turntype(idx) == baldr::Turn::Type::kRight ||
+        edge->turntype(idx) == baldr::Turn::Type::kSharpRight) {
+      seconds *= edge->stopimpact(idx);
+      is_turn = true;
+    }
 
+    // Apply density factor and stop impact penalty if there isn't traffic on this edge or you're not
+    // using traffic
+    if (!has_measured_speed) {
```

Comment:
Why don't we use the predecessor label's `has_measure_speed()` instead of passing-in the value explicitly here (similar to whats done in `TransitionCost()` ?

Reply:
the predecessor EdgeLabel is in reverse direction of actual travel. I imagine the 2 directed edges in opposing directions could have different speeds / measured speed flags.

Reply:
ah ok, that makes sense. thx @dnesbitt61

Reply:
@mandeepsandhu due to this https://github.com/valhalla/valhalla/issues/2896